core/txpool/legacypool/legacypool.go

```diff
@@ -679,6 +680,14 @@ func (pool *LegacyPool) add(tx *types.Transaction, local bool) (replaced bool, e
 	// the sender is marked as local previously, treat it as the local transaction.
 	isLocal := local || pool.locals.containsTx(tx)
 
+	// When the guardian module is enabled, check if the sender or recipient in the
+	// transaction is in the filter file
+	if instance := guardian.GetInstance(); instance != nil {
+		if instance.CheckTransaction(pool.signer, tx) {
+			validTxMeter.Mark(1)
```

Comment:
we should not use this metric since it's for valid transaction.
I suggest to create a new metric like`sanctionedTx`

Reply:
Done